scripts/Common.ps1

```diff
@@ -386,4 +386,746 @@ function Create-StorageContainer
     # get a storage context
     $storageContext = New-AzureStorageContext -StorageAccountName $StorageAccountName -StorageAccountKey $StorageAccountKey
     New-AzureStorageContainer -Name $StorageContainerName -Context $storageContext
-}
\ No newline at end of file
+}
+#################################################################
+#
+# Wrappers for All Azure Cmdlet Calls
+#
+#################################################################
+
+# Command, Activity, Parameters
+function Execute-AzureCommand
+{
+    param( [Parameter(Mandatory=$true)][hashtable]$InputParameters )
+
+    # we will make one generic call for every wrapped Azure Cmdlet
+    # With that approach, we unify the call pattern, retries & error handling
+    # special error handling will still be the responsibility of the caller
+
+    # the object we will return
+    $response = $null;
+
+    # we support individual function using custom maximum retries
+    # right now, they are not all enabled (but wired to do so)
+    [int]$MaxRetries = 3;
+    if ($InputParameters.ContainsKey("MaxRetries"))
+    {
+        try
+        {
+            $MaxRetries = [int]$InputParameters['MaxRetries'];
+        }
+        catch{ }
+    }
+
+    # check the expected exceptions
+    if ($InputParameters.ContainsKey('ExpectedException') -eq $false)
+    {
+        $InputParameters['ExpectedException'] = "";
+    }
+
+    # track the retries
+    $retryAttempt = 1;
+    while ($retryAttempt -le $MaxRetries)
+    {
+        try
+        {
+            Log-Message "Attempt [$($retryAttempt)|$($MaxRetries)] - $($InputParameters['Activity']) started." -Context $Context;
+
+            # handle the commands appropriately
+            switch ($InputParameters['Command']) 
+            {
+
+                "Find-AzureRmResource"
+                {
+                    if ($InputParameters['params'] -ne $null)
+                    {
+                         $response= Find-AzureRmResource -ResourceGroupNameContains $InputParameters['params'].ResourceGroupNameEquals -ResourceType $InputParameters['params'].ResourceType -Verbose ;  
+                    }
+                    break;
+                }
+                "Get-AzureRmLoadBalancer"
+                {
+                    if ($InputParameters['Name'] -ne $null)
+                    {
+                        $response= Get-AzureRmLoadBalancer -Name $InputParameters['Name'] -ResourceGroupName $InputParameters['ResourceGroup'] -Verbose ;  
+                    }
+                    break;
+                }
+
+                "Get-AzureRmLoadBalancerRuleConfig"
+                {
+                    if ($InputParameters['Name'] -ne $null)
+                    {
+                        $response= Get-AzureRmLoadBalancerRuleConfig -LoadBalancer $InputParameters['Name'];
+                    }
+                    break;
+                }
+
+                 "Remove-AzureRmLoadBalancerRuleConfig"
```

Comment:
nit; alignment. make sure all options are aligned correctly with the braces

Reply:
done.